### ISSUE TITLE
Web Accessibility for Screen Reader Users

### ISSUE THREAD

**Issue by chigkim** [nextcloud/server#8167]
It seems like a lot of controls can't be accessed by screen readers an keyboard.
For example, I can't open admin, profile, and other apps.
Tab and shift(or option)+tab should be able to focus on all the elements and enter to activate.
For Mac, you can test with VoiceOver command+f5.
For Windows, you can test with NVDA from nvaccess.org.
Could you look into this?
Thanks,

**Comment by MorrisJobke** [comment]
@nextcloud/designers @jancborchardt Do you know who looked some months ago into accessibility things?

**Comment by skjnldsv** [comment]
@pixelipo @MorrisJobke I would love to work on this and create a list of the accessibility stuff needed to be done! I already started to work on this with mobile/tablet/desktop rewriting and Contrast issue, I would mind creating a big issue with multiple pr fixing everything ! This needs to be done :)

**Comment by jancborchardt** [comment]
cc @nextcloud/accessibility :)

@chigkim are you up for contributing more or giving more detail what can be improved? That would be very helpful. As @pixelipo mentioned we did a lot of accessibility work but often things fall under the radar since none of us are experts.

**Comment by chigkim** [comment]
Thanks for looking into this.
I think best way might be for someone from development team to try to 
use nextcloud web interface using one of the screen readers I mentioned.
Also Chrome browser has a free extension called Chromevox on Chrome web 
store.
I think the priority should be to make all the elements focussable with 
tab and ship+tab, and activate with enter.
Also if there are clickable items with only graphics, it needs alt tag 
for screen reader to recognize the name of the icons.
Otherwise, screen reader might says the file name like a83jjsialr.png.
Here are some resources in markdown format.

# Web Content Accessibility Guidelines (WCAG)
* [Checklist by Teams](http://accessibility.voxmedia.com/)
* [Summary](https://www.w3.org/WAI/WCAG20/glance/)
* [CheckList](https://www.w3.org/WAI/WCAG20/quickref/)
* [WebAIM Checklist](http://webaim.org/standards/wcag/checklist)
* [Technique](https://www.w3.org/TR/WCAG20-TECHS/general.html)
* [Alt Text](http://webaim.org/techniques/alttext/)
* [Skip Links](http://webaim.org/techniques/skipnav/)

# Evaluation
* [Wave](http://wave.webaim.org/)
* [Voiceover](http://webaim.org/articles/voiceover/)

# Accessibility of Rich Internet Application (ARIA)
* [Aria](https://www.w3.org/TR/using-aria/)
* [WebAIM Aria](http://webaim.org/techniques/aria/)
* [WebAIM Aria Cause and 
Solution](http://webaim.org/blog/aria-cause-solution/)
* [Layout](https://www.w3.org/TR/html5/sections.html)
* [Main 
Element](https://www.w3.org/TR/html5/grouping-content.html#the-main-element)
* [Navigation Role](https://www.w3.org/TR/wai-aria/roles#navigation)
* [Expanded 
State](https://www.w3.org/TR/wai-aria/states_and_properties#aria-expanded)

Hope that helps!

**Comment by chigkim** [comment]
I'm wondering if there's any update on this.
I updated to  13.0.2, and I still can't access my settings without going to directly to index.php/settings/admin.
What's the direct URL where I can enable the app to mount external storage?

**Comment by tyrylu** [comment]
That's strange, i can access the admin settings screen just fine with the latest NVDA snapshot and Firefox 60 on Windows 10. You just click the last link in the header which expands a submenu from which you can go to settings or apps management. Yes, the label for the link is somewhat broken in 13.0.2 and an expandable aria attribute is also missing, but at least, it is a link, which is unfortunately not a 100% standard in today's web development.
And the direct link for the disabled apps section is: https://nextcloud.yourdomain.com/settings/apps?category=disabled#

**Comment by chigkim** [comment]
Thanks! That's good to know. I managed to get that to work using NVDA as 
you described.
For some reason, both Jaws and Narrator skip the link to expand and show 
settings, apps, and so on. Maybe because there's no text associated with 
the link?

**Comment by tyrylu** [comment]
Yes, it might (and very likely is) the cause. Fortunately, it is alerady fixed in the master branch, so it should get to a release sooner or later, someone knows the release schedule likely better than me, so a clarification would help there.

**Comment by jancborchardt** [comment]
@chigkim yeah, as @tyrylu pointed out, this should be fixed with https://github.com/nextcloud/server/pull/6700

More accessibility pointers are always very very welcome! :) Most helpful is to directly point out concrete issues which are happening (even better if they have pointers to solutions) – as we do go through Nextcloud with these tools, but not daily, and stuff of course also slips through.

**Comment by jancborchardt** [comment]
There’s also a new pull request up for review for keyboard accessibility of the app-navigation: https://github.com/nextcloud/server/pull/9951

And another one for high-contrast theme and font, as well as dark mode: https://github.com/nextcloud/server/pull/9862

And the pull request mentioned before which fixes header accessibility https://github.com/nextcloud/server/pull/6700

Let’s close this issue because it’s very broad and a lot has been fixed. And we’ll of course continue working on improving accessibility. What would be very helpful is to open separate issues for any problems you still encounter. Thanks a lot! :tada:
